### PR TITLE
Two Korean classification datasets added

### PR DESCRIPTION
**Pull request by dokato**
There are 82 million speakers of Korean (acc to wiki https://en.wikipedia.org/wiki/List_of_languages_by_total_number_of_speakers), yet there was only one dedicated classification task. This adds two classification:
- sarcasm detection: https://huggingface.co/datasets/kor_sarcasm
- tweet hate classification: https://huggingface.co/datasets/kor_hate 

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: docs/mmteb/points/670.jsonl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "dokato", "New dataset": 4}
+{"GitHub": "?", "Review PR": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/670.jsonl:2]:
```suggestion
{"GitHub": "KennethEnevoldsen", "Review PR": 2}
```

File: mteb/tasks/Classification/kor/KorHateClassification.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        bibtex_citation="""
+        @misc{moon2020beep,
+            title={BEEP! Korean Corpus of Online News Comments for Toxic Speech Detection}, 
+            author={Jihyung Moon and Won Ik Cho and Junbum Lee},
+            year={2020},
+            eprint={2005.12503},
+            archivePrefix={arXiv},
+            primaryClass={cs.CL}
+        }""",
+        n_samples={"train": 7896},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/kor/KorHateClassification.py:46]: I would downsample to 2048 (see suggested function above)

File: mteb/tasks/Classification/kor/KorSarcasmClassification.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        @misc{kim2019kocasm,
+            author = {Kim, Jiwon and Cho, Won Ik},
+            title = {Kocasm: Korean Automatic Sarcasm Detection},
+            year = {2019},
+            publisher = {GitHub},
+            journal = {GitHub repository},
+            howpublished = {https://github.com/SpellOnYou/korean-sarcasm}
+        }
+        """,
+        n_samples={"train": 9000},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/kor/KorSarcasmClassification.py:48]: same as above

**Review by KennethEnevoldsen — APPROVED** [review]
Everything looks good. I would downsample the dataset, but otherwise I believe it is fine

**Comment by dokato** [comment]
I fixed it, so it uses `self.stratified_subsampling` now, thanks for suggestions @KennethEnevoldsen

**Comment by KennethEnevoldsen** [comment]
Thanks @dokato! I have added you as a contributor to MTEB as well

**Pull request merged**